reference/src/representation/vectors.md

```diff
@@ -0,0 +1,93 @@
+# Representation of vectors
+
+**Disclaimer:** This chapter represents the consensus from issue
+[#38]. The statements in here are not (yet) "guaranteed"
+not to change until an RFC ratifies them.
+
+[#38]: https://github.com/rust-rfcs/unsafe-code-guidelines/issues/38
+
+Rust currently exposes vector types like `__m128` to users, but it does not
+expose a way for users to construct their own vector types.
+
+The set of currently-exposed vector types is _implementation-defined_ and it is
+currently different for each architecture.
+
+## Vector types
+
+Vector types are `repr(simd)` homogeneous tuple-structs containing `N` elements of type `T`:
+
+```rust
+#[repr(simd)]
+struct Vector<T, N>(T_0, ..., T_(N - 1));
+```
+
+The set of supported values of `T` and `N` is _implementation-defined_.
+
+The size of `Vector` is `N * size_of::<T>()` and its alignment is an
+_implementation-defined_ function of `T` and `N` greater than or equal to
+`align_of::<T>()`. That is:
+
+```rust
+assert_eq!(size_of::<Vector<T, N>>(), size_of::<T>() * N);
+assert!(align_of::<Vector<T, N>>() >= align_of::<T>());
+```
+
+That is, two distinct `repr(simd)` vector types that have the same `T` and the
+same `N` have the same size and alignment.
+
+The layout of elements within a vector is the same as that of `[T; N]` such that
+there is a a 1:1 correspondence between the indices of the vector and arrays:
```

Comment:
Nit: ABI alignment is often considered to be a part of "layout", but alignment can be different here (which affects type punning through pointers, though unions are fine). So for clarify I'd prefer to mention that problem explicitly and word the element order issue a bit differently, e.g., "elements are laid out in source field order, enabling random access to vector elements by reinterpreting the vector as an array: [example]".

Reply:
I've pushed a commit with your wording.